src/machines/java/com/enderio/machines/common/blockentity/base/PoweredMachineEntity.java

```diff
@@ -271,6 +271,7 @@ public void invalidateCaps() {
 
     @Override
     public void saveAdditional(CompoundTag pTag) {
+        //serialize internal field, don't fix it or you'll break solar panels
```

Comment:
Okay; I think a solution to this instead of dirty accessing and overloading could be to separate getEnergyStorage into two; getEnergyStorage <- this will be the same as the internal field; and getExposedEnergyStorage <- whatever wrapped output you want.
This will mean the behaviour is more clear, while still keeping solar panels working.

I'm happy to take this on as part of some of the machine reworking I have upcoming though, so if you would like we can deal with that there.

Reply:
I'll do that